draft-ietf-quic-transport.md

```diff
@@ -1168,35 +1168,85 @@ Destination and Source Connection ID fields match the Source and Destination
-version negotiation, a client MUST increase the packet number it uses on every
-packet it sends.  Packets MUST continue to use long headers ({{long-header}})
-and MUST include the new negotiated protocol version.
+selects an acceptable protocol version from the list provided by the server.  If
+the client does not support any of the versions the server offers, it aborts the
+connection attempt.
+
+The client then attempts to establish a connection using the version it selects.
+Packets MUST continue to use long headers ({{long-header}}) and MUST include the
+new negotiated protocol version.
+
+A Version Negotiation packet reveals which of the versions supported by the
+client are not supported by the server.  The client MUST add all versions the
+server does not support to the `unsupported_versions` field that is included
+with its transport parameters (see {{transport-parameter-versions}}), or the
+equivalent field in the version it selects.  The `unsupported_versions` field is
+used to validate the version negotiation process (see {{version-validation}}).
 
 The client MUST use the long header format and include its selected version on
 all packets until it has 1-RTT keys and it has received a packet from the server
 which is not a Version Negotiation packet.
 
 A client MUST NOT change the version it uses unless it is in response to a
-Version Negotiation packet from the server.  Once a client receives a packet
-from the server which is not a Version Negotiation packet, it MUST discard other
-Version Negotiation packets on the same connection.  Similarly, a client MUST
-ignore a Version Negotiation packet if it has already received and acted on a
-Version Negotiation packet.
-
-A client MUST ignore a Version Negotiation packet that lists the client's chosen
-version.
-
-A client MAY attempt 0-RTT after receiving a Version Negotiation packet.  A
-client that sends additional 0-RTT packets MUST NOT reset the packet number to 0
-as a result, see {{retry-0rtt-pn}}.
-
-Version negotiation packets have no cryptographic protection. The result of the
-negotiation MUST be revalidated as part of the cryptographic handshake (see
-{{version-validation}}).
+Version Negotiation packet from the server, or if the server picks a different,
+but compatible version (see {{version-upgrade}}).  Once a client receives a
+packet from the server which is not a Version Negotiation packet, it MUST
+discard other Version Negotiation packets on the same connection.  Similarly, a
+client MUST ignore a Version Negotiation packet if it has already received and
+acted on a Version Negotiation packet.
+
+A client MAY attempt 0-RTT after receiving a Version Negotiation packet.
+
+The format of the packet that a client sends might be different in the new
+version.  In this case, the client generates a new packet that conforms to the
+selected version.  Though different versions might convey information about
```

Comment:
Aren't we requiring the client to always regenerate a packet, now that `unsupported_versions` change from an empty list to a non-empty list when receiving a VN?

Reply:
Ahh, I missed that one.  I squashed the other text, but missed that one.

Reply:
Yeah! It's a gift to proof-readers :+1: